build.gradle.kts

```diff
@@ -0,0 +1,23 @@
+import org.jetbrains.kotlin.gradle.targets.js.nodejs.NodeJsRootExtension
+import org.jetbrains.kotlin.gradle.targets.js.npm.tasks.KotlinNpmInstallTask
+
+plugins {
+    alias(libs.plugins.kotlinx.binaryCompatibilityValidator)
+}
+
+val deploy by tasks.creating() {
+    dependsOn(getTasksByName("publish", true))
+    dependsOn(getTasksByName("publishNpm", true))
+}
+
+// Right now it is used for switching nodejs version which is supports generated wasm bytecode
+extensions.findByType(NodeJsRootExtension::class.java)?.let {
+    // canary nodejs that supports recent Wasm GC changes
+    it.nodeVersion = "21.0.0-v8-canary202309167e82ab1fa2"
+    it.nodeDownloadBaseUrl = "https://nodejs.org/download/v8-canary"
+}
+
+// We need to ignore unsupported engines (i.e. canary) for npm
```

Comment:
```suggestion
// We need to ignore unsupported engines (i.e. canary) for npm  - Remove after updating to Kotlin 2.0
```

Reply:
@adam-enko I'm not sure what the plan is for 2.0, but I'd release an intermediate version with 1.9.24, which allows to get rid of canary anyway.

Reply:
Thanks for the tip @lppedd! It certainly would be nice to be able to remove this npm-nightly config.

Reply:
Discussed with @dkrasnoff  - we'll tidy up the K/JS config in a later PR.